spring-core/src/main/java/org/springframework/core/convert/support/GenericConversionService.java

```diff
@@ -166,6 +169,44 @@ public boolean canBypassConvert(@Nullable TypeDescriptor sourceType, TypeDescrip
 		return (converter == NO_OP_CONVERTER);
 	}
 
+	@SuppressWarnings("unchecked")
+	@Nullable
+	public <S, T> List<T> convertList(@Nullable List<S> sourceList, Class<T> targetClass) {
+		return (List<T>) convertCollection(sourceList, TypeDescriptor.valueOf(targetClass), ArrayList::new);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Nullable
+	public <S, T> List<T> convertList(@Nullable List<S> sourceList, TypeDescriptor targetType) {
+		return (List<T>) convertCollection(sourceList, targetType, ArrayList::new);
+	}
+
+	@SuppressWarnings("unchecked")
```

Comment:
Also, after the "C extends Collection<T>", check if the  @SupressWarnings("unchecked") are still required